src/garage/torch/modules/cnn_module.py

```diff
@@ -0,0 +1,355 @@
+            raise ValueError(msg.format(str(var), n_layers, var))
+        return [copy.deepcopy(var) for _ in range(n_layers)]
+
+    @classmethod
+    def _check_valid_padding(cls, paddings, filter_sizes):
+        """Check if padding shape of convolutional layers are valid for `same`\
+        padding (zero padding) implementation.
+
+        Args:
+            paddings (tuple[int]): paddings
+            filter_sizes (tuple[int]): filter sizes
+
+        Raises:
+            ValueError: if the padding size is not equal to the expected shape
+
+        """
+        for index, (padding,
+                    filter_size) in enumerate(zip(paddings, filter_sizes)):
+            if padding != 0:  # only check when padding value is not 0
+                expected_padding = int((filter_size - 1) / 2)  # P = (F - 1) /2
+                assert (padding == expected_padding
+                        ), 'Invalid padding shape at index {}.\
+                Expected to be {}, but got {} instead'.format(
+                            index, expected_padding, padding)
+
+    # pylint: disable=arguments-differ
+    def forward(self, input_val):
+        """Forward method.
+
+        Args:
+            input_val (torch.Tensor): Input values with (N, C, H, W)
+                shape.
+
+        Returns:
+            List[torch.Tensor]: Output values
+
+        """
+        x = input_val
+        for layer in self._layers:
+            x = layer(x)
+        x = flatten(x)
+        return x
+
+
+def _conv(in_channels,
+          out_channels,
+          kernel_size,
+          stride=1,
+          padding=0,
+          padding_mode='zeros',
+          dilation=1,
+          bias=True):
+    """Helper function for performing convolution.
+
+    Args:
+        in_channels (int):
+            Number of channels in the input image
+        out_channels (int):
+            Number of channels produced by the convolution
+        kernel_size (int or tuple):
+            Size of the convolving kernel
+        stride (int or tuple): Stride of the convolution.
+            Default: 1
+        padding (int or tuple): Zero-padding added to both sides
+            of the input. Default: 0
+        padding_mode (string): 'zeros', 'reflect', 'replicate'
+            or 'circular'. Default: 'zeros'
+        dilation (int or tuple): Spacing between kernel elements.
+            Default: 1
+        bias (bool): If True, adds a learnable bias to the output.
+            Default: True
+
+    Returns:
+        torch.Tensor: The output of the 2D convolution.
+
+    """
+    return nn.Conv2d(in_channels=in_channels,
+                     out_channels=out_channels,
+                     kernel_size=kernel_size,
+                     stride=stride,
+                     padding=padding,
+                     padding_mode=padding_mode,
+                     dilation=dilation,
+                     bias=bias)
+
+
+# pylint: disable=missing-return-type-doc
+def flatten(x):
```

Comment:
please put this in garage.torch._functions (it will need a more descriptive name)